controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -271,7 +246,9 @@ public void triggerChange(ApplicationId applicationId, Change change) {
     public void cancelChange(ApplicationId applicationId) {
         applications().lockOrThrow(applicationId, application -> {
             deploymentQueue.removeJobs(application.id());
-            applications().store(application.withChange(Change.empty()));
+            applications().store(application.withChange(application.change().application()
```

Comment:
Looks like application changes can no longer be cancelled via the application API? I think that should still be supported.

Reply:
Fair point.

Reply:
Ok, I wanted to improve the "finishing of change" anyway, and that fits well with this -- both would require a "cancel this component of change" mechanism (and the former would also require a "is this part of change deployed" mechanism. I'll implement.

Reply:
I delayed detecting and cancelling individual parts of Change, and just allowed all change to be cancelled from the web API for now.